rct229/rules/section6/section6rule3.py

```diff
@@ -0,0 +1,93 @@
+            list_path="ruleset_model_instances[0].buildings[*].building_segments[*]",
+        )
+
+    class BuildingSegmentRule(RuleDefinitionListIndexedBase):
+        def __init__(self):
+            super(Section6Rule3.BuildingSegmentRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(True, False, True),
+                each_rule=Section6Rule3.BuildingSegmentRule.SpaceRule(),
+                index_rmr="baseline",
+                list_path="zones[*].spaces[*]",
+            )
+
+        def create_data(self, context, data=None):
+            building_segment_p = context.proposed
+
+            return {
+                "building_segment_lighting_status_type_dict_p": get_building_segment_lighting_status_type_dict(
+                    building_segment_p
+                )
+            }
+
+        class SpaceRule(RuleDefinitionBase):
+            def __init__(self):
+                super(
+                    Section6Rule3.BuildingSegmentRule.SpaceRule,
+                    self,
+                ).__init__(
+                    fail_msg="Lighting exists or is submitted with design documents. Lighting power density in P_RMR "
+                             "does not match U_RMR.",
+                    manual_check_required_msg = "Lighting is not yet designed, or lighting is as-designed or "
+                                                "as-existing but matches Table 9.5.1. Lighting power density in P_RMR"
+                                                " does not match U_RMR.",
+                    rmrs_used=UserBaselineProposedVals(True, False, True)
+                )
+
+            def get_calc_vals(self, context, data=None):
+                space_u = context.user
+                space_p = context.proposed
+                total_space_lpd_u = sum(
+                    find_all("interior_lighting[*].power_per_area", space_u)
+                )
+                total_space_lpd_p = sum(
+                    find_all("interior_lighting[*].power_per_area", space_p)
+                )
+
+                space_lighting_status_type_p = data[
+                    "building_segment_lighting_status_type_dict_p"
+                ][space_p["id"]]
+
+                return {
+                    "total_space_lpd_u": total_space_lpd_u,
+                    "total_space_lpd_p": total_space_lpd_p,
+                    "space_lighting_status_type_p": space_lighting_status_type_p,
+                }
+
+            def manual_check_required(self, context, calc_vals=None, data=None):
+                space_lighting_status_type_p = calc_vals["space_lighting_status_type_p"]
+                total_space_lpd_u = calc_vals["total_space_lpd_u"]
+                total_space_lpd_p = calc_vals["total_space_lpd_p"]
+                return not std_equal(total_space_lpd_u, total_space_lpd_p) and space_lighting_status_type_p is not LightingStatusType.AS_DESIGNED_OR_AS_EXISTING
```

Comment:
I believe the second ```not``` should be deleted. Could you verify this? because the RMD says ```space_lighting_status_type_dict[space.id] == "AS-DESIGNED OR AS-EXISTING" )```.

Reply:
I think this logic is correct.

Reply:
The logic statement here is to rule out the case 3 before `rule_check` is called. So the first part is `total_space_LPD_p != total_space_LPD_u` and the second part is: `space_lighting_status_type_dict[space.id] != "AS-DESIGNED OR AS-EXISTING"`. When both satisfied, then the manual check is required (undetermined)